app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -791,10 +791,18 @@ private void setInitialData(final long pid, final String title) {
         this.name = !TextUtils.isEmpty(title) ? title : "";
     }
 
-    private void setVideoCount(final long count) {
+    private void setVideoCountAndOverallDuration(final ArrayList<LocalItem> itemsList) {
         if (activity != null && headerBinding != null) {
-            headerBinding.playlistStreamCount.setText(Localization
-                    .localizeStreamCount(activity, count));
+            final long videoCount = itemsList.size();
+            final long playlistOverallDurationSeconds = itemsList.stream().mapToLong(x ->
```

Comment:
I would first filter by `PlaylistStreamEntry.class::isInstance`, then map with `PlaylistStreamEntry.class::cast`, then map to `PlaylistStreamEntry::getStreamEntity`, finally mapToLong `StreamEntryOrWhatever::getDuration` and sum.